livre/lib/cache/HistogramObject.cpp

```diff
@@ -58,10 +58,64 @@ void binData( const SRC_TYPE* rawData,
                                          compCount * j * dataBlockSize.z() +
                                          compCount * k + c;
                     const double data = rawData[ index ];
-                    const size_t binIndex =
-                            std::is_unsigned< SRC_TYPE >::value ?
-                            ( data - minVal ) / range * ( binCount - 1 ):
-                            (( data / range ) + 0.5f ) * ( binCount - 1 );
+                    const size_t binIndex =  histogram.getMin() + ( data - histogram.getMin( )) /
```

Comment:
Add an assert to ensure than range is divisible by binCount and try not to use floating point operations for integral types. Assuming that range is divisible by binCount that should be fairly easy.
Besides that, the code is not correct for signed types, e.g. int8 and a value of -100 produce this:
-128 + ( -100 - -127) / 255 * 255 = -128 + 27/255*255 = -128 + 27 =  -101 which when converted to size_t will produce an interesting result.